### PR TITLE
Fix for cleanup job failure when configmap not present

### PR DESCRIPTION
**Pull request by joejulian**
**What type of PR is this?**

bug

**What this PR does/ why we need it**:

Cleanup job fails if configmap doesn't exist. Don't fail, we didn't want it to exist anyway.

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-72743

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Review by branden — APPROVED** [review]
LGTM. I wonder what's deleting the configmap before this job can 🤔

**Review by gracedo — APPROVED** [review]
thanks, good call

File: staging/prometheus-operator/Chart.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ sources:
   - https://github.com/coreos/kube-prometheus
   - https://github.com/coreos/prometheus-operator
   - https://coreos.com/operators/prometheus
-version: 9.3.1
+version: 9.3.2
 appVersion: 0.38.1
 tillerVersion: ">=2.12.0"
```
Comment (gracedo) [inline staging/prometheus-operator/Chart.yaml:17]: lint is failing on 
```
Error validating data '/charts/staging/prometheus-operator/Chart.yaml' with schema '/etc/ct/chart_schema.yaml'
	tillerVersion: Unexpected element
```

File: staging/prometheus-operator/templates/prometheus/rules/mesosphere-rules/etcd-rules.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
+{{- if and .Values.mesosphereResources.create .Values.mesosphereResources.rules.etcd }}
 # Rules for alerting against etcd, without the etcd alerts: etcdHighNumberOfFailedGRPCRequests
 # Which at the moment are false positives at a rate of 100%
-{{- if and .Values.mesosphereResources.create .Values.mesosphereResources.rules.etcd }}
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
-  name: {{ printf "%s-%s" (include "prometheus-operator.fullname" .) "etcd" | trunc 63 | trimSuffix "-" }}
+  name: {{ printf "%s-%s" "etcd" (include "prometheus-operator.fullname" .) | trunc 63 | trimSuffix "-" }}
```
Comment (dkoshkin) [inline staging/prometheus-operator/templates/prometheus/rules/mesosphere-rules/etcd-rules.yaml:7]: Just so I understand, is this related to the issue?

**Review by dkoshkin — APPROVED** [review]

File: staging/prometheus-operator/templates/prometheus/rules/mesosphere-rules/etcd-rules.yaml (hunk 2 above)
Comment (joejulian) [reply staging/prometheus-operator/templates/prometheus/rules/mesosphere-rules/etcd-rules.yaml:7]: Gah, no. This was me testing something. Thanks for catching it.

Comment (joejulian) [reply staging/prometheus-operator/templates/prometheus/rules/mesosphere-rules/etcd-rules.yaml:7]: It probably should have been this way, though. Otherwise the unique part of the name could, conceivably, be pushed out past the truncate.

**Comment by gracedo** [comment]
I was just trying to understand why this was happening since the cleanup is done as a `pre-delete` but the error was happening during addon deployment, and dug into the logs - I found this error
`2020-10-27T03:40:08.888Z	ERROR	kubeaddons-controller.Addon.Helm3.install	failed helm install	{"Addon": "prometheus", "namespace": "kubeaddons", "chart": "prometheus-operator", "addon": "prometheus", "version": "9.3.1", "valuesCRC32": "7c166bd6", "error": "failed post-install: Internal error occurred: failed calling webhook \"webhook.federation.kommander.mesosphere.io\": Post https://kommander-federation-webhook.kommander.svc:443/validations?timeout=30s: dial tcp 10.0.58.126:443: connect: connection refused"}`

which then caused the addon to start getting deleted
```
2020-10-27T03:40:11.555Z	INFO	kubeaddons-controller.Addon.Helm3.Status	addon status result	{"Addon": "prometheus", "namespace": "kubeaddons", "chart": "prometheus-operator", "addon": "prometheus", "version": "9.3.1", "valuesCRC32": "7c166bd6", "status": {"ready":true,"stage":"failed"}}
2020-10-27T03:40:11.555Z	INFO	kubeaddons-controller.Addon.Helm3.Cleanup	helm deleting addon	{"Addon": "prometheus", "namespace": "kubeaddons", "chart": "prometheus-operator", "addon": "prometheus", "version": "9.3.1", "valuesCRC32": "7c166bd6"}
2020-10-27T03:40:11.555Z	INFO	kubeaddons-controller.Addon.Helm3.helmDelete	Deleting...	{"Addon": "prometheus", "namespace": "kubeaddons", "chart": "prometheus-operator", "addon": "prometheus", "version": "9.3.1", "valuesCRC32": "7c166bd6", "release": "prometheus-kubeaddons"}
```

**Comment by dkoshkin** [comment]
@gracedo are you saying this PR wouldn't be enough to fix it?

**Comment by gracedo** [comment]
I'm not sure if `Internal error occurred: failed calling webhook \"webhook.federation.kommander.mesosphere.io\": Post https://kommander-federation-webhook.kommander.svc:443/validations?timeout=30s: dial tcp 10.0.58.126:443: connect: connection refused` is a real persisting issue, or if by the time prom successfully deleted/reinstalled the endpoint would be reachable.. I'm looking into this service because we did make some changes to it recently to fix metrics, and I don't recall us hitting this in the past

**Comment by gracedo** [comment]
In any case, that would be a bug in yakcl, so we needn't block this PR on it
